cruise-control/src/main/java/com/linkedin/kafka/cruisecontrol/analyzer/goals/ReplicaDistributionGoal.java

```diff
@@ -121,9 +118,23 @@ public String name() {
   @Override
   protected void initGoalState(ClusterModel clusterModel, OptimizationOptions optimizationOptions) {
     super.initGoalState(clusterModel, optimizationOptions);
-    for (Broker broker : clusterModel.brokers()) {
-      BrokerAndSortedReplicas bas = new BrokerAndSortedReplicas(broker, broker.replicaComparator(DISK));
-      _brokerAndReplicasMap.put(broker.id(), bas);
+    clusterModel.trackSortedReplicas(name(),
```

Comment:
With this patch, `Broker#replicaComparator(Resource)` seems to be redundant. Do we intend to use it for other purposes in the future or can we safely remove it and continue relying on track/untrack sortedReplicas?

Reply:
I think we can remove it.